project/PythonClientPostBuild.scala

```diff
@@ -0,0 +1,140 @@
+   *
+   * This method copies custom files into the output directory and updates the version information.
+   *
+   * @param log                 The logger to output informational messages.
+   * @param openApiOutputDir    The directory where the OpenAPI generator outputs the files.
+   * @param baseDir             The base directory of the project.
+   * @param version             The version string from sbt.
+   */
+  def processGeneratedFiles(
+      log: Logger,
+      openApiOutputDir: String,
+      baseDir: File,
+      version: String
+  ): Unit = {
+
+    /**
+     * Copies a custom file to the output directory, overwriting if it exists.
+     *
+     * @param source       The path to the custom file to copy from.
+     * @param target       The path to the location to copy to.
+     * @param description  A description of the file being copied, used in log messages.
+     */
+    def copyFile(source: Path, target: Path, description: String): Unit = {
+      if (Files.exists(source)) {
+        Files.copy(source, target, StandardCopyOption.REPLACE_EXISTING)
+        log.info(s"Copied $description to ${target}.")
+      } else {
+        sys.error(s"Required $description not found! Expected at: ${source.toAbsolutePath}")
+      }
+    }
+
+    /**
+     * Updates the version in the specified file.
+     *
+     * @param filePath  The path to the file to update.
+     * @param version   The version string to set.
+     */
+    def updateVersionInFile(filePath: Path, version: String): Unit = {
+      if (Files.exists(filePath)) {
+        val content = Files.readAllLines(filePath, StandardCharsets.UTF_8)
+        val updatedContent = content.toArray.map {
+          case line: String =>
+            if (line.trim.startsWith("version =")) {
+              s"""version = "$version""""
+            } else if (line.trim.startsWith("    version=")) {
+              s"""    version="$version",""""
+            } else {
+              line
+            }
+        }
+        Files.write(filePath, updatedContent.mkString("\n").getBytes(StandardCharsets.UTF_8))
+        log.info(s"Updated version in ${filePath.getFileName} to $version")
+      } else {
+        sys.error(s"File not found: ${filePath.toAbsolutePath}")
+      }
+    }
+
+    val pythonVersion = convertVersionForPython(version)
+
+    val buildDir = baseDir.toPath.resolve("build")
+    val clientPyproject = buildDir.resolve("pyproject.toml")
+    val clientSetupPy = buildDir.resolve("setup.py")
+    val projectReadme = buildDir.resolve("README.md")
+
+    val generatedPyproject = Paths.get(openApiOutputDir, "pyproject.toml")
+    val generatedSetupPy = Paths.get(openApiOutputDir, "setup.py")
+    val generatedReadme = Paths.get(openApiOutputDir, "README.md")
+
+    copyFile(clientPyproject, generatedPyproject, "pyproject.toml")
+    copyFile(clientSetupPy, generatedSetupPy, "setup.py")
+    copyFile(projectReadme, generatedReadme, "README.md")
```

Comment:
this along with copyFile function seems like a bit of bloated, repeated code. why not just make it loop 
```
Seq("pyproject.toml", ...).foreach { name =>
   Files.copy(
     buildDir.resolve(name), 
     Paths.get(openApiOutputDir, name), 
    StandardCopyOption.REPLACE_EXISTING)  
}
```

Reply:
I think I'm remembering how to write functional scala now ;) how's this update?